flink-connectors/flink-connector-kinesis/pom.xml

```diff
@@ -56,6 +58,18 @@ under the License.
 		</dependency>
 
 		<dependency>
+		    <groupId>org.apache.httpcomponents</groupId>
+		    <artifactId>httpclient</artifactId>
+		    <version>${apache.httpclient.version}</version>
+		</dependency>
+
+		<dependency>
+		    <groupId>org.apache.httpcomponents</groupId>
+		    <artifactId>httpcore</artifactId>
+		    <version>${apache.httpcore.version}</version>
+		</dependency>
```

Comment:
For `promoteTransitiveDependencies`, what does it really mean? https://maven.apache.org/plugins/maven-shade-plugin/shade-mojo.html#promoteTransitiveDependencies is kind of very confusing to understand.

To your second point, my intention is neither adding http-components 4.3.x for hadoop-aws, nor for AWS s3. It is actually for aws-java-sdk-kinesis used by flink-connector-kinesis. This way, users won't worry about versions of http-components depended by aws-java-sdk-kinesis anymore. Does it make sense?